lib/src/picture_provider.dart

```diff
@@ -460,9 +461,11 @@ abstract class AssetBundlePictureProvider
   PictureInfoDecoder<String> decoder;
 
   @override
-  set currentColor(Color? color) {
-    _currentColor = color;
-    decoder = decoderBuilder(_currentColor);
+  set theme(SvgTheme? theme) {
+    _theme = theme;
```

Comment:
We should guard this with a check about whether the theme == the other theme or not.